### PR TITLE
Sid nick usc

### PR DESCRIPTION
**Pull request by sidd328**
# Overview

UCSB Subject controller works with GET all and POST endpoints (along with unit testing)

### CODE REVIEW

File: src/main/java/edu/ucsb/cs156/team02/controllers/UCSBSubjectController.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+@Slf4j
+public class UCSBSubjectController extends ApiController {
+    @Autowired
+    UCSBSubjectRepository subjectRepository;
+
+    @Autowired
+    ObjectMapper mapper;
+
+    @ApiOperation(value = "List all UCSB Subjects")
+    @GetMapping("/all")
```
Comment (LucienLuc) [inline src/main/java/edu/ucsb/cs156/team02/controllers/UCSBSubjectController.java:42]: If you want both Users and Admins to be able to use this API add 
`@PreAuthorize("hasRole('ROLE_USER') || hasRole('ROLE_ADMIN')")`

(same for all other endpoints)

File: src/main/java/edu/ucsb/cs156/team02/controllers/UCSBSubjectController.java [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        subject.setRelatedDeptCode(relatedDeptCode);
+        subject.setInactive(inactive);
+        UCSBSubject savedSubject = subjectRepository.save(subject);
+        return savedSubject;
+    }
+
+    @ApiOperation(value = "Get a single UCSB Subject by ID if it is in the databse.")
+    @GetMapping("")
+    public ResponseEntity<String> getSubjectById(
+            @ApiParam("id") @RequestParam Long id) throws JsonProcessingException {
```
Comment (LucienLuc) [inline src/main/java/edu/ucsb/cs156/team02/controllers/UCSBSubjectController.java]: Add tests for this method.

File: src/test/java/edu/ucsb/cs156/team02/controllers/UCSBSubjectControllerTests.java [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+    @MockBean
+    UserRepository userRepository;
+
+    // Authorization tests for /api/todos/admin/all
+
+    @Test
+    public void ucsb_subject_get_logged_out() throws Exception {
+        mockMvc.perform(get("/api/UCSBSubjects/all"))
+                .andExpect(status().isOk());
+    }
```
Comment (LucienLuc) [inline src/test/java/edu/ucsb/cs156/team02/controllers/UCSBSubjectControllerTests.java]: Maybe add mocks and assertions to actually check the return value of this endpoint.

**Review by LucienLuc — CHANGES_REQUESTED** [review]

File: src/main/java/edu/ucsb/cs156/team02/controllers/UCSBSubjectController.java (hunk 1 above)
Comment (sidd328) [reply src/main/java/edu/ucsb/cs156/team02/controllers/UCSBSubjectController.java:42]: Fixed

File: src/main/java/edu/ucsb/cs156/team02/controllers/UCSBSubjectController.java (hunk 2 above)
Comment (sidd328) [reply src/main/java/edu/ucsb/cs156/team02/controllers/UCSBSubjectController.java]: Fixed

File: src/test/java/edu/ucsb/cs156/team02/controllers/UCSBSubjectControllerTests.java (hunk 3 above)
Comment (sidd328) [reply src/test/java/edu/ucsb/cs156/team02/controllers/UCSBSubjectControllerTests.java]: Done

**Review by LucienLuc — APPROVED** [review]
LGTM!

**Pull request merged**